### PR TITLE
Updates for Qt6

### PR DESCRIPTION
**Pull request by EricAtORS**
This is different from #1023 in 2 ways:
1. You can compile with Qt6 from this changelist.
2. QRegExp` is taken from the `Qt5CoreCompat` rather than being aliased to QRegularExpression, minimizing the changes.

Just some additional notes:
I've only tested it for CTKCore and CTKWidgets.
I'm also using Conan to manage the compilation and so not all Qt macros are there.
This is something I'm looking to bring up with the conan maintainers.

I am migrating to Qt6, and so as a first pass, I want to see that the CI will build for older versions as well.

### CODE REVIEW

File: Applications/Testing/Cpp/CMakeLists.txt [hunk 1]
```diff
@@ -32,8 +32,10 @@ if(CTK_APP_ctkDICOMQuery AND CTK_APP_ctkDICOMRetrieve)
   ctk_add_executable_utf8(ctkDICOMApplicationTest1 ctkDICOMApplicationTest1.cpp)
   if(CTK_QT_VERSION VERSION_LESS "5")
     target_link_libraries(ctkDICOMApplicationTest1 ${QT_LIBRARIES})
-  else()
+  elsei  if(CTK_QT_VERSION VERSION_LESS "6")
```
Comment (jamesobutler) [inline Applications/Testing/Cpp/CMakeLists.txt:35]: @EricAtORS typo here causing build errors in the status checks

File: CMake/ctkMacroBuildPlugin.cmake [hunk 2]
```diff
@@ -316,16 +316,16 @@ macro(ctkMacroBuildPlugin)
...
-    if(CMAKE_VERSION VERSION_LESS 2.8.12)
-      target_link_libraries(${lib_name} Qt5::Test)
-    else()
-      target_link_libraries(${lib_name} PRIVATE Qt5::Test)
-    endif()
+      find_package(Qt${CTK_QT_VERSION} Test REQUIRED)
+      if(CMAKE_VERSION VERSION_LESS 2.8.12)
+        target_link_libraries(${lib_name} Qt::Test)
+      else()
+        target_link_libraries(${lib_name} PRIVATE Qt::Test)
```
[the remark this answers is not in the document]
Comment (EricAtORS) [reply CMake/ctkMacroBuildPlugin.cmake:327]: I didn't know that it was only for qt 5.15+ for the versionless targets. I'll fix them.

File: Libs/DICOM/Core/ctkDICOMDatabase.h [hunk 3]
```diff
@@ -363,7 +363,7 @@ class CTK_DICOM_CORE_EXPORT ctkDICOMDatabase : public QObject
   /// Insert lists of tags into the cache as a batch query operation
   Q_INVOKABLE bool cacheTags (const QStringList sopInstanceUIDs, const QStringList tags, const QStringList values);
   /// Remove all tags corresponding to a SOP instance UID
-  Q_INVOKABLE void removeCachedTags(const QString sopInstanceUID);
+  void removeCachedTags(const QString sopInstanceUID);
```
Comment (jamesobutler) [inline Libs/DICOM/Core/ctkDICOMDatabase.h:366]: This appears to be reverting recent commit https://github.com/commontk/CTK/commit/8f349212143bf23469a7e4c4f08617423a23538c. This diff here should be reverted to maintain that commit's changes.

File: Utilities/DGraph/CMakeLists.txt [hunk 4]
```diff
@@ -18,7 +18,7 @@
 #
 ###########################################################################
 
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 2.8)
```
Comment (jamesobutler) [inline Utilities/DGraph/CMakeLists.txt:21]: This appears to be reverting recent commit https://github.com/commontk/CTK/commit/87b18d82a7898a5461d84c2146cc0fa60f1e1ba1. This diff here should be reverted to maintain that commit's changes.

File: CMake/ctkMacroBuildPlugin.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (jamesobutler) [reply CMake/ctkMacroBuildPlugin.cmake:327]: I see you've made some changes with the versioned target, though I still see this specific place in the code and some other places with only a versionless target being used like `Qt::`

File: CMake/ctkMacroSetupQt.cmake.orig [hunk 5]
```diff
@@ -0,0 +1,92 @@
+###########################################################################
```
Comment (jamesobutler) [inline CMake/ctkMacroSetupQt.cmake.orig:1]: Do you still need this file? Appears to be an artifact leftover when you were changing things in ctkMacroSetupQt.cmake, so you should be able to remove this file.

**Comment by EricAtORS** [comment]
I don't understand why it fails with Qt5. For now, I'm going to close the PR.

**Pull request closed without merging**